msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/ltw/TestCase2582292.java

```diff
@@ -0,0 +1,150 @@
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.client.msal.automationapp.testpass.broker.ltw;
+
+import androidx.test.uiautomator.UiObjectNotFoundException;
+
+import com.microsoft.identity.client.msal.automationapp.R;
+import com.microsoft.identity.client.msal.automationapp.testpass.broker.AbstractMsalBrokerTest;
+import com.microsoft.identity.client.ui.automation.annotations.LTWTests;
+import com.microsoft.identity.client.ui.automation.annotations.SupportedBrokers;
+import com.microsoft.identity.client.ui.automation.app.MsalTestApp;
+import com.microsoft.identity.client.ui.automation.broker.BrokerLTW;
+import com.microsoft.identity.client.ui.automation.broker.BrokerMicrosoftAuthenticator;
+import com.microsoft.identity.client.ui.automation.interaction.PromptParameter;
+import com.microsoft.identity.client.ui.automation.interaction.microsoftsts.MicrosoftStsPromptHandlerParameters;
+import com.microsoft.identity.labapi.utilities.client.ILabAccount;
+import com.microsoft.identity.labapi.utilities.client.LabQuery;
+import com.microsoft.identity.labapi.utilities.constants.AzureEnvironment;
+import com.microsoft.identity.labapi.utilities.constants.TempUserType;
+import com.microsoft.identity.labapi.utilities.constants.UserRole;
+import com.microsoft.identity.labapi.utilities.constants.UserType;
+import com.microsoft.identity.labapi.utilities.exception.LabApiException;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.List;
+
+// If LTW is the active broker, and request is made through Authenticator from an old MSAL in shared device mode, nothing should break
+// https://identitydivision.visualstudio.com/Engineering/_workitems/edit/2582292
+@LTWTests
+@SupportedBrokers(brokers = {BrokerLTW.class})
+public class TestCase2582292 extends AbstractMsalBrokerTest {
+
+    @Test
+    public void test_2582292() throws LabApiException, InterruptedException, UiObjectNotFoundException {
+        final String username = mLabAccount.getUsername();
+        final String password = mLabAccount.getPassword();
+
+        // Install new Auth app with broker SDK changes of broker selection logic
+        final BrokerMicrosoftAuthenticator brokerMicrosoftAuthenticator = new BrokerMicrosoftAuthenticator();
+        brokerMicrosoftAuthenticator.install();
+
+        // Install legacy MSAL Test app (Msal test app with no broker selection logic)
+        final MsalTestApp msalTestApp = new MsalTestApp();
+        msalTestApp.installOldApk();
```

Comment:
move these steps to below line 70. This would keep msalTestApp steps in 1 place

Reply:
Sure, but if I move the steps, it would be against the order of test steps listed here: https://identitydivision.visualstudio.com/Engineering/_workitems/edit/2582292
Do you want me to move the steps in the test case and change the order of the steps listed above as well?